requests_hardened/client.py

```diff
@@ -30,14 +30,12 @@ def send(self, request: PreparedRequest, **kwargs: Any) -> Response:
 
     def prepare_request(self, request: Request) -> PreparedRequest:
         url = request.url
+        headers = request.headers or {}
+        # Cast potentially immutable header list to `dict`
+        if not isinstance(headers, dict):
+            headers = cast(dict, dict(**headers))
```

Comment:
Now we can move it back under the block `if self._config.ip_filter_enable is True:` 😅